File: vendas-services/src/main/java/br/com/vendas/services/customer/CustomerService.java
package br.com.vendas.services.customer;

import java.math.BigDecimal;
import java.util.List;

import br.com.vendas.domain.customer.Customer;
import br.com.vendas.exception.ApplicationException;
import br.com.vendas.services.support.ServiceResponse;

public interface CustomerService {
	
	/**
	 * Retorna todos os clientes de determinada empresa
	 * @param organizationID
	 * @return
	 */
	ServiceResponse<List<Customer>> findAllByOrganizationID(Integer organizationID, Integer branchID, Integer offset);
	
	/**
	 * Salva o cliente
	 * @param customer
	 * @return
	 */
	ServiceResponse<Customer> save(Integer userID, Customer customer);
	
 	/**
 	 * Retorna todos os produtos que iniciem com a descrição, ou código do produto, ou código de barras passada no filter.
 	 * @param filter
 	 * @param organizationID
 	 * @param branchID
 	 * @param offset
 	 * @return
 	 */
 	ServiceResponse<List<Customer>> findByIDOrNameOrEmail(String filter,Integer organizationID, Integer branchID, Integer offset, Integer limit);
 	
 	/**
 	 * Retorna o limite de crédito do cliente
 	 * @param customerID
 	 * @return
 	 * @throws ApplicationException 
 	 */
 	ServiceResponse<BigDecimal> getAvaliableCreditLimit(Integer customerID) throws ApplicationException;
 	
 	/**
 	 * Retorna o cliente por id
 	 * @param id
 	 * @return
 	 */
 	ServiceResponse<Customer> findByID(Integer id);
 	
	/**
	 * Tem pagamento vencido ?
	 * @param customerID
	 * @return
	 */
	ServiceResponse<Boolean> hasExpiratePayment(Integer customerID);
	
	
	
	void save( List<Customer> customers );
}

File: vendas-web/src/main/webapp/app/js/ng/services/CustomerService.js
'use strict';

vendasApp.factory('CustomerService',function(Restangular,UtilityService, LocalStorageService){
	var _customer;
	//Cliente que sera editado
	var customerEdition;
	return {
		
		
		/**
		 * Registra um novo cliente para o sistema.
		 * @param fields
		 * @returns
		 */
		save : function(customer){
			var newCustomer = Restangular.all('v1').all('customer').all('saveCustomer').post(customer);
			return newCustomer;
        },


 

        /**
         * Retorna todos os clientes de determinada empresa.
         * @returns
         */
        getAllByOrganization: function(organizationID,branchID,offset){
        	
        	var parameters = {
					'organizationID' : organizationID, 
					'branchID': branchID,
					'offset' : offset					
					};
        	
        	
        	return Restangular.all('v1').all("customer").all("getCustomersByOrganizationID").getList(parameters).then(function(result){
				var p = {};
				p.value = result.value;
				p.rowCount = result.rowCount;			
				return p;
			});
        },
        
        /**
		 * Retorna todos os produtos de determinada filial.
		 * @returns
		 */
		getAllByFilter : function(filter,organizationID,branchID, offset){
			
			var parameters = {
					'filter': filter,
					'organizationID' : organizationID, 
					'branchID' : branchID,
					'offset': offset
			};


			return Restangular.all("v1").all("customer").all("getAllByFilter").getList(parameters).then(function(result){
				var p = {};
				p.value = result.value;
				p.code = result.code;
				p.message = result.message;
				p.rowCount = result.rowCount;
				return p;
			});
		},
		
		/**
		 * Retorna o limite de crédito do cliente
		 */
		getAvaliableCreditLimit : function(customerID){
			
			var parameters = {
					'customerID': customerID
			};

			return Restangular.all("v1").all("customer").all("getAvaliableCreditLimit").getList(parameters).then(function(result){
				var p = {};
				p.value = result.value;
				p.code = result.code;
				p.message = result.message;
				p.rowCount = result.rowCount;
				return p;
			});
		},
		
		/**
		 * Verifica se existe algum pagamento vencido
		 */
		hasExpiratePayment : function(customerID){
			
			var parameters = {
					'customerID': customerID
			};

			return Restangular.all("v1").all("customer").all("hasExpiratePayment").getList(parameters).then(function(result){
				var p = {};
				p.value = result.value;
				p.code = result.code;
				p.message = result.message;
				p.rowCount = result.rowCount;
				return p;
			});
		},
        
        
        setCustomerEdition: function(customer){
        	customerEdition = customer;
        },
        
        getCustomerEdition: function(){
        	return customerEdition;
        },        
	};	
});
